packages/emnapi/src/value/create.ts

```diff
@@ -323,14 +331,25 @@ function napi_create_external_buffer (
 // @ts-expect-error
 ): napi_status {
   $PREAMBLE!(env, (envObject) => {
-    const status = _napi_create_external_arraybuffer(env, data, length, finalize_cb, finalize_hint, result)
+    const status = _emnapi_create_memory_view(
+      env,
+      napi_typedarray_type.napi_uint8_array,
+      data,
+      length,
+      finalize_cb,
+      finalize_hint,
+      result
+    )
     if (status !== napi_status.napi_ok) {
       return status
     }
     const handleId = $makeGetValue('result', 0, '*')
     const handle = emnapiCtx.handleStore.get(handleId)!
+    const info = emnapiExternalMemory.wasmMemoryViewTable.get(handle.value)!
+    const buffer = Buffer.from(HEAPU8.buffer, info.ptr, info.length)
```

Comment:
Maybe it would be a bit easier to add `napi_typedarray_type.napi_buffer` to the enum?

After all, `Buffer` is more or less another TypedArray and could be created and managed from the same place instead of creating an intermediate `Uint8Array` first.

Reply:
Might be also worth updating `napi_create_buffer` the way you suggested in the same PR.

Reply:
Done.